Validation/SiTrackerPhase2V/interface/TrackerPhase2ValidationUtil.h

```diff
@@ -1,12 +1,54 @@
 #ifndef _Validation_SiTrackerPhase2V_TrackerPhase2ValidationUtil_h
 #define _Validation_SiTrackerPhase2V_TrackerPhase2ValidationUtil_h
+
 #include "SimDataFormats/TrackingHit/interface/PSimHit.h"
 #include "SimDataFormats/Track/interface/SimTrack.h"
+#include "SimDataFormats/TrackingAnalysis/interface/TrackingParticle.h"
+#include <cmath>
+#include <tuple>  // for std::tuple
 
 namespace phase2tkutil {
 
   bool isPrimary(const SimTrack& simTrk, const PSimHit* simHit);
   static constexpr float cmtomicron = 1e4;
 
+  inline std::tuple<float, float, float> computeZ0LxyD0(const TrackingParticle& tp, float pt, float eta, float phi) {
+    float vz = tp.vz();
+    float vx = tp.vx();
+    float vy = tp.vy();
+    float charge = tp.charge();
+
+    float t = std::tan(2.0 * std::atan(1.0) - 2.0 * std::atan(std::exp(-eta)));
+    float delx = -vx;
+    float dely = -vy;
+    float K = 0.01 * 0.5696 / pt * charge;  // curvature correction
+    float A = 1. / (2. * K);
+    float x0p = delx - A * std::sin(phi);
+    float y0p = dely + A * std::cos(phi);
+    const float pi = 4.0 * std::atan(1.0);
+    float delphi = phi - std::atan2(-K * x0p, K * y0p);
```

Comment:
Just #include DataFormats/Math/interface/deltaPhi.h and then do float deltaphi = reco::deltaPhi(phi, std::atan2(-K*x0p, K*y0p). Then it is automatically in the range -PI to +PI.